src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -70,13 +68,31 @@ public static JaegerSpan ToJaegerSpan(this Activity activity)
                 ProcessActivityTagRef);
 
             string peerServiceName = null;
-            if ((activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer) && jaegerTags.PeerService != null)
+            if (activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer)
             {
-                // Send peer.service for remote calls.
-                peerServiceName = jaegerTags.PeerService;
+                // If priority = 0 that means peer.service may have already been included in tags
+                var addPeerServiceTag = jaegerTags.PeerServicePriority > 0;
 
-                // If priority = 0 that means peer.service was already included in tags.
-                if (jaegerTags.PeerServicePriority > 0)
+                var hostNameOrIpAddress = jaegerTags.HostName ?? jaegerTags.IpAddress;
+
+                // peer.service has not already been included, but net.peer.name/ip and optionally net.peer.port are present
+                if ((jaegerTags.PeerService == null || jaegerTags.PeerServicePriority > 0)
```

Comment:
You could replace `jaegerTags.PeerServicePriority > 0` here with `addPeerServiceTag` since you basically cached that result into a local variable.

Reply:
good catch